core/src/main/scala/com/karumi/shot/reports/ConsoleReporter.scala

```diff
@@ -0,0 +1,65 @@
+package com.karumi.shot.reports
+
+import com.karumi.shot.base64.Base64Encoder
+import com.karumi.shot.domain._
+import com.karumi.shot.ui.Console
+
+class ConsoleReporter(console: Console) {
+
+  def showBase64Error(comparision: ScreenshotsComparisionResult,
+                      outputFolder: String): Unit = {
+    console.show(
+      "🤖  The option printBase64 is enabled. In order to see the generated diff images, run the following commands in your terminal:")
+    console.lineBreak()
+    comparision.screenshots.foreach(screenshot => {
+      showScreenshotBase64Error(outputFolder, screenshot)
+    })
+  }
+
+  def showErrors(comparision: ScreenshotsComparisionResult): Unit = {
+    console.showError(
+      "❌  Hummmm...the following screenshot tests are broken:\n")
+    comparision.errors.foreach { error =>
+      error match {
+        case ScreenshotNotFound(screenshot) =>
+          console.showError(
+            "   🔎  Recorded screenshot not found for test: " + screenshot.name)
+        case DifferentScreenshots(screenshot) =>
+          console.showError(
+            "   🤔  The application UI has been modified for test: " + screenshot.name)
+          console.showError(
+            "            💾  You can find the original screenshot here: " + screenshot.recordedScreenshotPath)
+          console.showError(
+            "            🆕  You can find the new recorded screenshot here: " + screenshot.temporalScreenshotPath)
+        case DifferentImageDimensions(screenshot,
+                                      originalDimension,
+                                      newDimension) => {
+          console.showError(
+            "   📱  The size of the screenshot taken has changed for test: " + screenshot.name)
+          console.showError(
+            "            💾  Original screenshot dimension: " + originalDimension + ". You can find the original screenshot here: " + screenshot.recordedScreenshotPath)
+          console.showError(
+            "            🆕  New recorded screenshot dimension: " + newDimension + ". You can find the new recorded screenshot here: " + screenshot.temporalScreenshotPath)
+        }
+
+        case _ =>
+          console.showError(
+            "   😞  Ups! Something went wrong while comparing your screenshots but we couldn't identify the cause. If you think you've found a bug, please open an issue at https://github.com/karumi/shot.")
+      }
+      console.lineBreak()
+    }
+  }
+
+  private def showScreenshotBase64Error(outputFolder: String,
```

Comment:
I'd move the base64 generation to the class generating the screenshot comparison result error. The diff between the images and how we check if there are errors or not is implemented in parallel inside that class. This will avoid a performance issue.

Reply:
Do you mean that We should save on files the base64 results and get it back in the ConsoleReporter? 🤔 then, better it does it in another PR

Reply:
what I mean is that you can add the base 64 file to the error information during the image comparison as an optional field so you have a ``DifferentScreenshot`` case class with an ``Option[String]`` base 64. This forces you to move the base 64 calculus to the place where the error is created, helps to clean the console reporter and also improves the performance..